apis/placement/v1/clusterresourceplacement_types.go

```diff
@@ -0,0 +1,853 @@
+type PropertySelectorRequirement struct {
+	// Name is the name of the property; it should be a Kubernetes label name.
+	// +required
+	Name string `json:"name"`
+
+	// Operator specifies the relationship between a cluster's observed value of the specified
+	// property and the values given in the requirement.
+	// +required
+	Operator PropertySelectorOperator `json:"operator"`
+
+	// Values are a list of values of the specified property which Fleet will compare against
+	// the observed values of individual member clusters in accordance with the given
+	// operator.
+	//
+	// At this moment, each value should be a Kubernetes quantity. For more information, see
+	// https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Quantity.
+	//
+	// If the operator is Gt (greater than), Ge (greater than or equal to), Lt (less than),
+	// or `Le` (less than or equal to), Eq (equal to), or Ne (ne), exactly one value must be
+	// specified in the list.
+	//
+	// +kubebuilder:validation:MaxItems=1
+	// +required
+	Values []string `json:"values"`
+}
+
+// PropertySelector helps user specify property requirements when picking clusters for resource
+// placement.
+type PropertySelector struct {
+	// MatchExpressions is an array of PropertySelectorRequirements. The requirements are AND'd.
+	// +required
+	MatchExpressions []PropertySelectorRequirement `json:"matchExpressions"`
+}
+
+// PropertySorter helps user specify how to sort clusters based on a specific property.
+type PropertySorter struct {
+	// Name is the name of the property which Fleet sorts clusters by.
+	// +required
+	Name string `json:"name"`
+
+	// SortOrder explains how Fleet should perform the sort; specifically, whether Fleet should
+	// sort in ascending or descending order.
+	// +required
+	SortOrder PropertySortOrder `json:"sortOrder"`
+}
+
+type ClusterSelectorTerm struct {
+	// LabelSelector is a label query over all the joined member clusters. Clusters matching
+	// the query are selected.
+	//
+	// If you specify both label and property selectors in the same term, the results are AND'd.
+	// +optional
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+
+	// PropertySelector is a property query over all joined member clusters. Clusters matching
+	// the query are selected.
+	//
+	// If you specify both label and property selectors in the same term, the results are AND'd.
+	//
+	// At this moment, PropertySelector can only be used with
+	// `RequiredDuringSchedulingIgnoredDuringExecution` affinity terms.
+	// +optional
+	PropertySelector *PropertySelector `json:"propertySelector,omitempty"`
+
+	// PropertySorter sorts all matching clusters by a specific property and assigns different weights
+	// to each cluster based on their observed property values.
+	//
+	// At this moment, PropertySorter can only be used with
+	// `PreferredDuringSchedulingIgnoredDuringExecution` affinity terms.
+	// +optional
+	PropertySorter *PropertySorter `json:"propertySorter,omitempty"`
```

Comment:
these two are preview

Reply:
Per earlier discussions, a beta-level hint comment is added.